### PR TITLE
New tls branch

### PR DESCRIPTION
**Pull request by nivshani**

### CODE REVIEW

File: netlab/L5/tls_protocol_layer.hpp [hunk 1]
```diff
@@ -529,6 +531,7 @@ namespace netlab {
 						Extension extension(it, it + sizeof(Extension));
 						this->handshake.body.clientHello.extensions_union.extensions.push_back(extension);
 						it += sizeof(Extension);
+						i+= sizeof(Extension);
```
Comment (GabrielKlyatis) [inline netlab/L5/tls_protocol_layer.hpp]: Why do you need the second increment?
EDIT: assuming cause you deserialize 2 bytes, see comment above

File: netlab/L5/tls_protocol_layer.hpp [hunk 2]
```diff
@@ -520,7 +522,7 @@ namespace netlab {
 					this->handshake.body.clientHello.compression_methods[i] = static_cast<CompressionMethod>(*it);
 				}
 				// Extensions
-				extensions_length = *it;
+				extensions_length = deserialize_2_bytes(it);
```
Comment (GabrielKlyatis) [inline netlab/L5/tls_protocol_layer.hpp:525]: From what I recall, extensions length is 1 byte

File: netlab/L5/tls_protocol_layer.hpp [hunk 3]
```diff
@@ -556,30 +559,34 @@ namespace netlab {
...
+						i += extensions_length;
 					}
 				}
 				break;
 			case CERTIFICATE:
 				// Deserialize certificate struct
 				certificate_list_length = deserialize_3_bytes(it);
 				this->handshake.body.certificate.certificate_list.resize(certificate_list_length - CERTIFICATE_HANDSHAKE_OFFSET_LENGTH);
-				for (size_t i = 0; i < certificate_list_length - CERTIFICATE_HANDSHAKE_OFFSET_LENGTH; i++) {
+				for (size_t i = 0; i < certificate_list_length - CERTIFICATE_HANDSHAKE_OFFSET_LENGTH; ) {
```
Comment (GabrielKlyatis) [inline netlab/L5/tls_protocol_layer.hpp]: Just curious about the difference between incrementing i in the for loop statement in comparison to this style

**Review by GabrielKlyatis — COMMENTED** [review]
I left some comments about my original files, apart from that looks great to me
anyway, delete in destructor needs to be added to this merge.

File: netlab/L5/tls_protocol_layer.hpp (hunk 1 above)
Comment (nivshani) [reply netlab/L5/tls_protocol_layer.hpp]: no, in the for loop you have  (uint16_t i = 0; i < extensions_length; i++) 
but extensions_length is the total length in bytes and not the number ot extentions

File: netlab/L5/tls_protocol_layer.hpp (hunk 3 above)
Comment (nivshani) [reply netlab/L5/tls_protocol_layer.hpp]: you dont know how much to increament, cartificates are not fixed in size

File: netlab/L5/tls_protocol_layer.hpp (hunk 2 above)
Comment (nivshani) [reply netlab/L5/tls_protocol_layer.hpp:525]: 2 bytes

**Pull request merged**